modin/pandas/test/test_io.py

```diff
@@ -1291,6 +1306,21 @@ def test_to_csv_with_index(self):
         eval_to_file(modin_df, pandas_df, "to_csv", "csv")
 
 
+# Leave this test apart from the test classes, which skip the default to pandas
+# warning check. We want to make sure we are NOT defaulting to pandas for a
+# path relative to user home.
+# TODO(https://github.com/modin-project/modin/issues/3655): Get rid of this
+# commment once we turn all default to pandas messages into errors.
+def test_read_csv_relative_to_user_home(monkeypatch, make_csv_file):
+    unique_filename = get_unique_filename()
+
+    patch_expanduser("~/testfile.csv", unique_filename, monkeypatch)
```

Comment:
If we would ever change our implementation to stop using `os.path.expanduser()` but expand that `~` in some other way this test would break, but the implementation would be correct...

Why don't we change `HOME` environment variable to some temp directory for the duration of this test, and let that ~/testfile.csv` point to a file in that directory instead?

Reply:
Ah, I see that's a great idea because both windows and Posix should count `$HOME` as home. Done.